server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -49,21 +50,42 @@ protected boolean searchWithCollector(
         boolean hasFilterCollector,
         boolean hasTimeout
     ) throws IOException {
-        return searchWithCollectorManager(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
+        // create the top docs collector
+        final TopDocsCollectorContext topDocsFactory = createTopDocsCollectorContext(searchContext, hasFilterCollector);
+        return searchWithCollector(searchContext, searcher, query, collectors, topDocsFactory, hasFilterCollector, hasTimeout);
+    }
+
+    protected boolean searchWithCollector(
```

Comment:
Btw, I think this method should be part of the `DefaultQueryPhaseSearcher` right? (not only concurrent one)

Reply:
Oh sorry, `@Override` is missed

Reply:
```
@Override
protected boolean searchWithCollector(
```